src/actions/navigation.js

```diff
@@ -23,15 +25,13 @@ function willNavigate() {
  * @static
  */
 function navigated() {
-  return ({ dispatch }) => {
-    // We need to load all the sources again because they might have
-    // come from bfcache, so we won't get a `newSource` notification.
-    //
-    // TODO: This seems to be buggy on the debugger server side. When
-    // the page is loaded from bfcache, we still get sources from the
-    // *previous* page as well. For now, emulate the current debugger
-    // behavior by not showing sources loaded by bfcache.
-    // return dispatch(sources.loadSources());
+  return ({ dispatch, getState }: ThunkArgs) => {
+    setTimeout(() => {
+      if (getSources(getState()).size == 0) {
+        const threadClient = firefox.getThreadClient();
+        threadClient.getSources();
```

Comment:
Looking good.

1. does this fix the problem, my fear is that getSources will make one call for the sources, but it won't trigger `nweSource` by itself
2. i think we can refactor this code a bit to make it more consistent with `util/client`

https://gist.github.com/jasonLaster/de3bd2ecbacaf7518c42336e199d5137
https://gist.github.com/dde3f2cd62733746e562f9d6acd48e76
 
Goal:
1. add a new firefox client command for fetching sources
2. add a `timeout` helper so we can do it with async/await
3. call `newSources` in this action which i think we need to do